File: Programming Languages/4 languages in 2 styles/OO/Javascript/Javascript OO/ProcessCMDArgs.mjs
import {DocumentProcessorBuilder} from "./DocumentProcessorBuilder";

export function parseCMDArgs(cmdArgs) {

    function parseCommands(cmdArgs) {
        let secondCommand;
        if ((cmdArgs.length > 6)) {
            secondCommand = cmdArgs[6];
        }
        else if ((cmdArgs.length < 4)) {
            throw "too few arguments";
        }

        function fillObjectsToReturn(firstCommand, secondCommand) {
            const returnObject = new DocumentProcessorBuilder();
            if ((firstCommand === "grep")) {
                returnObject.setLineFilterer();
            }
            if (firstCommand === "wc" || secondCommand === "wc") {
                returnObject.setCaseConverter();
                returnObject.setWordFinder();
                returnObject.setNonABCFilterer();
                returnObject.setWordCounter();
            }
            return returnObject;
        }

        return fillObjectsToReturn(cmdArgs[2], secondCommand);
    }

    function parseArguments(cmdArgs, DPB) {
        let objectToReturn = [];
        objectToReturn.fileName = cmdArgs[3];
        if (DPB.LineFilterer !== undefined && DPB.LineFilterer != null) {
            objectToReturn.fileName = cmdArgs[4];
            objectToReturn.wordsToSearchFor = cmdArgs[3];
        }
        objectToReturn.DocumentProcessorBuilder = DPB;
        return objectToReturn;
    }

    const DPB = parseCommands(cmdArgs);
    return parseArguments(cmdArgs, DPB);
    
}

File: Programming Languages/4 languages in 2 styles/OO/Python/Python OO/ProcessCMDArgs.py
import DocumentProcessorBuilder


def parseCMDArgs(listOfCMDArgs):

    def parse_commands(listOfCMDArgs):
        second_command = None;
        if len(listOfCMDArgs) > 5 :
            second_command = listOfCMDArgs[5]
        elif len(listOfCMDArgs) < 3:
            raise RuntimeError("too few arguments")

        def fillObjectsToReturn (first_command, second_command):
            return_object = DocumentProcessorBuilder.DocumentProcessorBuilder()
            if first_command == "grep":
                return_object.set_line_filterer()
            if first_command == "wc" or second_command == "wc":
                return_object.set_case_converter()
                return_object.set_word_finder()
                return_object.set_non_abc_filterer()
                return_object.set_word_counter()
            return return_object

        return fillObjectsToReturn(listOfCMDArgs[1], second_command)

    def parse_arguments(listOfCMDArgs, dpb):
        file_name = listOfCMDArgs[2]
        if hasattr(dpb, 'line_filterer'):
            file_name = listOfCMDArgs[3]
            words_to_search_for = listOfCMDArgs[2]
            return {'file_name': file_name, 'words_to_search_for': words_to_search_for, 'dpb': dpb}
        dictionary_to_return = {'file_name': file_name, 'dpb':dpb}
        return dictionary_to_return

    dpb = parse_commands(listOfCMDArgs)

    return parse_arguments(listOfCMDArgs, dpb)
